src/main.cpp

```diff
@@ -840,23 +841,20 @@ bool AcceptToMemoryPoolWorker(CTxMemPool &pool,
         if (fLimitFree && nFees < ::minRelayTxFee.GetFee(nSize))
         {
             static double dFreeCount = 0;
+            static int64_t nLastFreeTime;
+            static int64_t nFreeLimit = GetArg("-limitfreerelay", 15);
+            static CCriticalSection csLimiter;
+            LOCK(csLimiter);
 
-            // Use an exponentially decaying ~10-minute window:
-            dFreeCount *= std::pow(1.0 - 1.0 / 600.0, (double)(nNow - nLastTime));
-            nLastTime = nNow;
-
-            // -limitfreerelay unit is thousand-bytes-per-minute
             // At default rate it would take over a month to fill 1GB
-            LOG(MEMPOOL, "Rate limit dFreeCount: %g => %g\n", dFreeCount, dFreeCount + nSize);
-            if ((dFreeCount + nSize) >= (nFreeLimit * 10 * 1000 * nLargestBlockSeen / BLOCKSTREAM_CORE_MAX_BLOCK_SIZE))
+            int64_t nFreeLimitScaled = nFreeLimit * nLargestBlockSeen / BLOCKSTREAM_CORE_MAX_BLOCK_SIZE;
+            if (RateLimitExceeded(dFreeCount, nLastFreeTime, nFreeLimitScaled, nSize))
             {
```

Comment:
We already calculated nFreeLimit previously.  We do this a little differently from everybody else and also use a exponential decay because we gradually choke off nFreeLimit depending on the size of the mempool.   What you've done here actually breaks our functionality in this regard.  The way it was before was how it should be working, you might want to consider reverting this part of your code.  We can discuss on slack if you like.  

Also I don't see anywhere, maybe i missed it, where you used -limitrespendrelay?

Reply:
@ptschip see `RelayLimiter` class definition in `src/respend/respendrelayer.cpp`:

```c++
+            int64_t limit = GetArg("-limitrespendrelay", DEFAULT_LIMITRESPENDRELAY);
+            if (RateLimitExceeded(respendCount, lastRespendTime, limit, size)) {
+                LOG(RESPEND, "respend: Double-spend relay rejected by rate limiter\n");
+                return true;
+            }